Release 3.1.0 (#1296)
microsoft/onefuzz: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## 3.1.0

> ### Fixed

Reviewer: As @bmc-msft suggested, it'd be nice to preserve consistent section ordering (when many are present). This is just alphabetical: "Added", "Changed", "Fixed".

> * Deployment: Fixed an issue with multitenant deployment caused by a mismatch between the identifier used to configure the app registration and value used to authenticate the CLI client. [#1270](https://github.com/microsoft/onefuzz/pull/1270)

Reviewer 2: multitenant ==> multi-tenant

Reviewer: ```suggestion
* Deployment: Fixed an issue with multi-tenant deployment caused by a mismatch between the identifier used to configure the app registration and value used to authenticate the CLI client. [#1270](https://github.com/microsoft/onefuzz/pull/1270)
```

* Service: Fixed `scaleset proxy reset` to reset all proxies in specified region. [#1275](https://github.com/microsoft/onefuzz/pull/1275)

> * CLI: Temporarily ignoring type errors from `azure-storage-blob` due to invalid python type signatures. [#1258](https://github.com/microsoft/onefuzz/pull/1258)

Reviewer: ```suggestion
* CLI: Temporarily ignore type errors from `azure-storage-blob` due to invalid python type signatures. [#1258](https://github.com/microsoft/onefuzz/pull/1258)
```

### Added
* Service: Added ability to configure virtual network IP ranges. [#1268](https://github.com/microsoft/onefuzz/pull/1268)

> * Deployment: Added `flake8` to the deployment process to align with rest of the python codebase linting process. [#1286](https://github.com/microsoft/onefuzz/pull/1286)

Reviewer: ```suggestion
* Deployment: Added `flake8` to the deployment process to align with rest of the Python codebase linting. [#1286](https://github.com/microsoft/onefuzz/pull/1286)
```

Reviewer: (I think this one got lost somehow when you rearranged the sections)

* Service: Added custom extensions to enable Microsoft Security Monitoring extensions. [#1184](https://github.com/microsoft/onefuzz/pull/1184)

> * CLI: Added `--readonly_inputs` option for `onefuzz templates libfuzzer` command. [#1247](https://github.com/microsoft/onefuzz/pull/1247)

Reviewer: ```suggestion
* CLI: Added `--readonly_inputs` option to the `libfuzzer_basic` template. [#1247](https://github.com/microsoft/onefuzz/pull/1247)
```

* Service: Added ability to configure virtual network IP ranges. [#1268](https://github.com/microsoft/onefuzz/pull/1268)
* Deployment: Added `flake8` to the deployment process to align with rest of the python codebase linting process. [#1286](https://github.com/microsoft/onefuzz/pull/1286)
* Service: Added custom extensions to enable Microsoft Security Monitoring extensions. [#1184](https://github.com/microsoft/onefuzz/pull/1184)

> * CLI: Added `--readonly_inputs` option to the `libfuzzer_basic` template. [#1247](https://github.com/microsoft/onefuzz/pull/1247)

Reviewer: Sorry, my original suggestion had a typo! 😅 
```suggestion
* CLI: Added `--readonly_inputs` option to the `libfuzzer basic` template. [#1247](https://github.com/microsoft/onefuzz/pull/1247)
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## 3.1.0
### Changed
* CLI: Increased the default verbosity of destructive CLI commands. [#1264](https://github.com/microsoft/onefuzz/pull/1264)
* Agent/Supervisor/Proxy: Updated multiple third-party Rust dependencies. [#1239](https://github.com/microsoft/onefuzz/pull/1239), [#1240](https://github.com/microsoft/onefuzz/pull/1240), [#1236](https://github.com/microsoft/onefuzz/pull/1236), [#1238](https://github.com/microsoft/onefuzz/pull/1238), [#1245](https://github.com/microsoft/onefuzz/pull/1245), [#1246](https://github.com/microsoft/onefuzz/pull/1246), [#1252](https://github.com/microsoft/onefuzz/pull/1252), [#1253](https://github.com/microsoft/onefuzz/pull/1253), [#1254](https://github.com/microsoft/onefuzz/pull/1254), [#1257](https://github.com/microsoft/onefuzz/pull/1257), [#1261](https://github.com/microsoft/onefuzz/pull/1261), [#1262](https://github.com/microsoft/onefuzz/pull/1262), [#1276](https://github.com/microsoft/onefuzz/pull/1276), [#1278](https://github.com/microsoft/onefuzz/pull/1278)

### Fixed

> * Deployment: Fixed deployment issues in the `westus3` region due to `Region doesn't support Classic resource mode for Application Insights resources` error.

Reviewer: ```suggestion
* Deployment: Fixed deployment in some regions by deploying newer, broadly-supported versions of Application Insights resources. [#1291](https://github.com/microsoft/onefuzz/pull/1291)
```
